{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'error_rate': {'ceiling': None,\n",
      "                 'colour': '55,126,184',\n",
      "                 'description': 'Error rate: mismatches (NM) / bases mapped '\n",
      "                                '(CIGAR)',\n",
      "                 'dmax': 100.0,\n",
      "                 'dmin': 0.0,\n",
      "                 'floor': None,\n",
      "                 'format': '{:,.2f}',\n",
      "                 'hidden': None,\n",
      "                 'max': 100,\n",
      "                 'min': 0,\n",
      "                 'minRange': None,\n",
      "                 'modify': 100.0,\n",
      "                 'namespace': 'Samtools',\n",
      "                 'placement': 1000.0,\n",
      "                 'rid': 'mqc-generalstats-samtools-error_rate',\n",
      "                 'scale': 'OrRd',\n",
      "                 'shared_key': None,\n",
      "                 'suffix': '%',\n",
      "                 'title': 'Error rate'},\n",
      "  'non-primary_alignments': {'ceiling': None,\n",
      "                             'colour': '55,126,184',\n",
      "                             'description': 'Non-primary alignments (millions)',\n",
      "                             'dmax': 2.277565,\n",
      "                             'dmin': 0.0,\n",
      "                             'floor': None,\n",
      "                             'format': '{:,.1f}',\n",
      "                             'hidden': None,\n",
      "                             'max': None,\n",
      "                             'min': 0,\n",
      "                             'minRange': None,\n",
      "                             'modify': 1e-06,\n",
      "                             'namespace': 'Samtools',\n",
      "                             'placement': 1000.0,\n",
      "                             'rid': 'mqc-generalstats-samtools-non_primary_alignments',\n",
      "                             'scale': 'PuBu',\n",
      "                             'shared_key': 'read_count',\n",
      "                             'title': 'M Non-Primary'},\n",
      "  'raw_total_sequences': {'ceiling': None,\n",
      "                          'colour': '55,126,184',\n",
      "                          'description': 'Total sequences in the bam file '\n",
      "                                         '(millions)',\n",
      "                          'dmax': 2.277565,\n",
      "                          'dmin': 0.0,\n",
      "                          'floor': None,\n",
      "                          'format': '{:,.1f}',\n",
      "                          'hidden': None,\n",
      "                          'max': None,\n",
      "                          'min': 0,\n",
      "                          'minRange': None,\n",
      "                          'modify': 1e-06,\n",
      "                          'namespace': 'Samtools',\n",
      "                          'placement': 1000.0,\n",
      "                          'rid': 'mqc-generalstats-samtools-raw_total_sequences',\n",
      "                          'scale': 'GnBu',\n",
      "                          'shared_key': 'read_count',\n",
      "                          'title': 'M Total seqs'},\n",
      "  'reads_MQ0_percent': {'ceiling': None,\n",
      "                        'colour': '55,126,184',\n",
      "                        'description': '% of Reads that are Ambiguously Placed '\n",
      "                                       '(MapQ=0)',\n",
      "                        'dmax': 100.0,\n",
      "                        'dmin': 0.0,\n",
      "                        'floor': None,\n",
      "                        'format': '{:,.1f}',\n",
      "                        'hidden': True,\n",
      "                        'max': 100,\n",
      "                        'min': 0,\n",
      "                        'minRange': None,\n",
      "                        'modify': None,\n",
      "                        'namespace': 'Samtools',\n",
      "                        'placement': 1000.0,\n",
      "                        'rid': 'mqc-generalstats-samtools-reads_MQ0_percent',\n",
      "                        'scale': 'OrRd',\n",
      "                        'shared_key': None,\n",
      "                        'suffix': '%',\n",
      "                        'title': '% MapQ 0 Reads'},\n",
      "  'reads_mapped': {'ceiling': None,\n",
      "                   'colour': '55,126,184',\n",
      "                   'description': 'Reads Mapped in the bam file (millions)',\n",
      "                   'dmax': 2.277565,\n",
      "                   'dmin': 0.0,\n",
      "                   'floor': None,\n",
      "                   'format': '{:,.1f}',\n",
      "                   'hidden': None,\n",
      "                   'max': None,\n",
      "                   'min': 0,\n",
      "                   'minRange': None,\n",
      "                   'modify': 1e-06,\n",
      "                   'namespace': 'Samtools',\n",
      "                   'placement': 1000.0,\n",
      "                   'rid': 'mqc-generalstats-samtools-reads_mapped',\n",
      "                   'scale': 'GnBu',\n",
      "                   'shared_key': 'read_count',\n",
      "                   'title': 'M Reads Mapped'},\n",
      "  'reads_mapped_percent': {'ceiling': None,\n",
      "                           'colour': '55,126,184',\n",
      "                           'description': '% Mapped Reads',\n",
      "                           'dmax': 100.0,\n",
      "                           'dmin': 0.0,\n",
      "                           'floor': None,\n",
      "                           'format': '{:,.1f}',\n",
      "                           'hidden': None,\n",
      "                           'max': 100,\n",
      "                           'min': 0,\n",
      "                           'minRange': None,\n",
      "                           'modify': None,\n",
      "                           'namespace': 'Samtools',\n",
      "                           'placement': 1000.0,\n",
      "                           'rid': 'mqc-generalstats-samtools-reads_mapped_percent',\n",
      "                           'scale': 'RdYlGn',\n",
      "                           'shared_key': None,\n",
      "                           'suffix': '%',\n",
      "                           'title': '% Mapped'},\n",
      "  'reads_properly_paired_percent': {'ceiling': None,\n",
      "                                    'colour': '55,126,184',\n",
      "                                    'description': '% Properly Paired Reads',\n",
      "                                    'dmax': 100.0,\n",
      "                                    'dmin': 0.0,\n",
      "                                    'floor': None,\n",
      "                                    'format': '{:,.1f}',\n",
      "                                    'hidden': False,\n",
      "                                    'max': 100,\n",
      "                                    'min': 0,\n",
      "                                    'minRange': None,\n",
      "                                    'modify': None,\n",
      "                                    'namespace': 'Samtools',\n",
      "                                    'placement': 1000.0,\n",
      "                                    'rid': 'mqc-generalstats-samtools-reads_properly_paired_percent',\n",
      "                                    'scale': 'RdYlGn',\n",
      "                                    'shared_key': None,\n",
      "                                    'suffix': '%',\n",
      "                                    'title': '% Proper Pairs'}},\n",
      " {'error_rate': {'ceiling': None,\n",
      "                 'colour': '55,126,184',\n",
      "                 'description': 'Error rate: mismatches (NM) / bases mapped '\n",
      "                                '(CIGAR)',\n",
      "                 'dmax': 100.0,\n",
      "                 'dmin': 0.0,\n",
      "                 'floor': None,\n",
      "                 'format': '{:,.2f}',\n",
      "                 'hidden': None,\n",
      "                 'max': 100,\n",
      "                 'min': 0,\n",
      "                 'minRange': None,\n",
      "                 'modify': 100.0,\n",
      "                 'namespace': 'Samtools',\n",
      "                 'placement': 1000.0,\n",
      "                 'rid': 'mqc-generalstats-samtools-error_rate',\n",
      "                 'scale': 'OrRd',\n",
      "                 'shared_key': None,\n",
      "                 'suffix': '%',\n",
      "                 'title': 'Error rate'},\n",
      "  'non-primary_alignments': {'ceiling': None,\n",
      "                             'colour': '55,126,184',\n",
      "                             'description': 'Non-primary alignments (millions)',\n",
      "                             'dmax': 2.558834,\n",
      "                             'dmin': 0.0,\n",
      "                             'floor': None,\n",
      "                             'format': '{:,.1f}',\n",
      "                             'hidden': None,\n",
      "                             'max': None,\n",
      "                             'min': 0,\n",
      "                             'minRange': None,\n",
      "                             'modify': 1e-06,\n",
      "                             'namespace': 'Samtools',\n",
      "                             'placement': 1000.0,\n",
      "                             'rid': 'mqc-generalstats-samtools-non_primary_alignments',\n",
      "                             'scale': 'PuBu',\n",
      "                             'shared_key': 'read_count',\n",
      "                             'title': 'M Non-Primary'},\n",
      "  'raw_total_sequences': {'ceiling': None,\n",
      "                          'colour': '55,126,184',\n",
      "                          'description': 'Total sequences in the bam file '\n",
      "                                         '(millions)',\n",
      "                          'dmax': 2.558834,\n",
      "                          'dmin': 0.0,\n",
      "                          'floor': None,\n",
      "                          'format': '{:,.1f}',\n",
      "                          'hidden': None,\n",
      "                          'max': None,\n",
      "                          'min': 0,\n",
      "                          'minRange': None,\n",
      "                          'modify': 1e-06,\n",
      "                          'namespace': 'Samtools',\n",
      "                          'placement': 1000.0,\n",
      "                          'rid': 'mqc-generalstats-samtools-raw_total_sequences',\n",
      "                          'scale': 'GnBu',\n",
      "                          'shared_key': 'read_count',\n",
      "                          'title': 'M Total seqs'},\n",
      "  'reads_MQ0_percent': {'ceiling': None,\n",
      "                        'colour': '55,126,184',\n",
      "                        'description': '% of Reads that are Ambiguously Placed '\n",
      "                                       '(MapQ=0)',\n",
      "                        'dmax': 100.0,\n",
      "                        'dmin': 0.0,\n",
      "                        'floor': None,\n",
      "                        'format': '{:,.1f}',\n",
      "                        'hidden': True,\n",
      "                        'max': 100,\n",
      "                        'min': 0,\n",
      "                        'minRange': None,\n",
      "                        'modify': None,\n",
      "                        'namespace': 'Samtools',\n",
      "                        'placement': 1000.0,\n",
      "                        'rid': 'mqc-generalstats-samtools-reads_MQ0_percent',\n",
      "                        'scale': 'OrRd',\n",
      "                        'shared_key': None,\n",
      "                        'suffix': '%',\n",
      "                        'title': '% MapQ 0 Reads'},\n",
      "  'reads_mapped': {'ceiling': None,\n",
      "                   'colour': '55,126,184',\n",
      "                   'description': 'Reads Mapped in the bam file (millions)',\n",
      "                   'dmax': 2.558834,\n",
      "                   'dmin': 0.0,\n",
      "                   'floor': None,\n",
      "                   'format': '{:,.1f}',\n",
      "                   'hidden': None,\n",
      "                   'max': None,\n",
      "                   'min': 0,\n",
      "                   'minRange': None,\n",
      "                   'modify': 1e-06,\n",
      "                   'namespace': 'Samtools',\n",
      "                   'placement': 1000.0,\n",
      "                   'rid': 'mqc-generalstats-samtools-reads_mapped',\n",
      "                   'scale': 'GnBu',\n",
      "                   'shared_key': 'read_count',\n",
      "                   'title': 'M Reads Mapped'},\n",
      "  'reads_mapped_percent': {'ceiling': None,\n",
      "                           'colour': '55,126,184',\n",
      "                           'description': '% Mapped Reads',\n",
      "                           'dmax': 100.0,\n",
      "                           'dmin': 0.0,\n",
      "                           'floor': None,\n",
      "                           'format': '{:,.1f}',\n",
      "                           'hidden': None,\n",
      "                           'max': 100,\n",
      "                           'min': 0,\n",
      "                           'minRange': None,\n",
      "                           'modify': None,\n",
      "                           'namespace': 'Samtools',\n",
      "                           'placement': 1000.0,\n",
      "                           'rid': 'mqc-generalstats-samtools-reads_mapped_percent',\n",
      "                           'scale': 'RdYlGn',\n",
      "                           'shared_key': None,\n",
      "                           'suffix': '%',\n",
      "                           'title': '% Mapped'},\n",
      "  'reads_properly_paired_percent': {'ceiling': None,\n",
      "                                    'colour': '55,126,184',\n",
      "                                    'description': '% Properly Paired Reads',\n",
      "                                    'dmax': 100.0,\n",
      "                                    'dmin': 0.0,\n",
      "                                    'floor': None,\n",
      "                                    'format': '{:,.1f}',\n",
      "                                    'hidden': False,\n",
      "                                    'max': 100,\n",
      "                                    'min': 0,\n",
      "                                    'minRange': None,\n",
      "                                    'modify': None,\n",
      "                                    'namespace': 'Samtools',\n",
      "                                    'placement': 1000.0,\n",
      "                                    'rid': 'mqc-generalstats-samtools-reads_properly_paired_percent',\n",
      "                                    'scale': 'RdYlGn',\n",
      "                                    'shared_key': None,\n",
      "                                    'suffix': '%',\n",
      "                                    'title': '% Proper Pairs'}},\n",
      " {'error_rate': {'ceiling': None,\n",
      "                 'colour': '55,126,184',\n",
      "                 'description': 'Error rate: mismatches (NM) / bases mapped '\n",
      "                                '(CIGAR)',\n",
      "                 'dmax': 100.0,\n",
      "                 'dmin': 0.0,\n",
      "                 'floor': None,\n",
      "                 'format': '{:,.2f}',\n",
      "                 'hidden': None,\n",
      "                 'max': 100,\n",
      "                 'min': 0,\n",
      "                 'minRange': None,\n",
      "                 'modify': 100.0,\n",
      "                 'namespace': 'Samtools',\n",
      "                 'placement': 1000.0,\n",
      "                 'rid': 'mqc-generalstats-samtools-error_rate',\n",
      "                 'scale': 'OrRd',\n",
      "                 'shared_key': None,\n",
      "                 'suffix': '%',\n",
      "                 'title': 'Error rate'},\n",
      "  'non-primary_alignments': {'ceiling': None,\n",
      "                             'colour': '55,126,184',\n",
      "                             'description': 'Non-primary alignments (millions)',\n",
      "                             'dmax': 3.032828,\n",
      "                             'dmin': 0.0,\n",
      "                             'floor': None,\n",
      "                             'format': '{:,.1f}',\n",
      "                             'hidden': None,\n",
      "                             'max': None,\n",
      "                             'min': 0,\n",
      "                             'minRange': None,\n",
      "                             'modify': 1e-06,\n",
      "                             'namespace': 'Samtools',\n",
      "                             'placement': 1000.0,\n",
      "                             'rid': 'mqc-generalstats-samtools-non_primary_alignments',\n",
      "                             'scale': 'PuBu',\n",
      "                             'shared_key': 'read_count',\n",
      "                             'title': 'M Non-Primary'},\n",
      "  'raw_total_sequences': {'ceiling': None,\n",
      "                          'colour': '55,126,184',\n",
      "                          'description': 'Total sequences in the bam file '\n",
      "                                         '(millions)',\n",
      "                          'dmax': 3.032828,\n",
      "                          'dmin': 0.0,\n",
      "                          'floor': None,\n",
      "                          'format': '{:,.1f}',\n",
      "                          'hidden': None,\n",
      "                          'max': None,\n",
      "                          'min': 0,\n",
      "                          'minRange': None,\n",
      "                          'modify': 1e-06,\n",
      "                          'namespace': 'Samtools',\n",
      "                          'placement': 1000.0,\n",
      "                          'rid': 'mqc-generalstats-samtools-raw_total_sequences',\n",
      "                          'scale': 'GnBu',\n",
      "                          'shared_key': 'read_count',\n",
      "                          'title': 'M Total seqs'},\n",
      "  'reads_MQ0_percent': {'ceiling': None,\n",
      "                        'colour': '55,126,184',\n",
      "                        'description': '% of Reads that are Ambiguously Placed '\n",
      "                                       '(MapQ=0)',\n",
      "                        'dmax': 100.0,\n",
      "                        'dmin': 0.0,\n",
      "                        'floor': None,\n",
      "                        'format': '{:,.1f}',\n",
      "                        'hidden': True,\n",
      "                        'max': 100,\n",
      "                        'min': 0,\n",
      "                        'minRange': None,\n",
      "                        'modify': None,\n",
      "                        'namespace': 'Samtools',\n",
      "                        'placement': 1000.0,\n",
      "                        'rid': 'mqc-generalstats-samtools-reads_MQ0_percent',\n",
      "                        'scale': 'OrRd',\n",
      "                        'shared_key': None,\n",
      "                        'suffix': '%',\n",
      "                        'title': '% MapQ 0 Reads'},\n",
      "  'reads_mapped': {'ceiling': None,\n",
      "                   'colour': '55,126,184',\n",
      "                   'description': 'Reads Mapped in the bam file (millions)',\n",
      "                   'dmax': 3.032828,\n",
      "                   'dmin': 0.0,\n",
      "                   'floor': None,\n",
      "                   'format': '{:,.1f}',\n",
      "                   'hidden': None,\n",
      "                   'max': None,\n",
      "                   'min': 0,\n",
      "                   'minRange': None,\n",
      "                   'modify': 1e-06,\n",
      "                   'namespace': 'Samtools',\n",
      "                   'placement': 1000.0,\n",
      "                   'rid': 'mqc-generalstats-samtools-reads_mapped',\n",
      "                   'scale': 'GnBu',\n",
      "                   'shared_key': 'read_count',\n",
      "                   'title': 'M Reads Mapped'},\n",
      "  'reads_mapped_percent': {'ceiling': None,\n",
      "                           'colour': '55,126,184',\n",
      "                           'description': '% Mapped Reads',\n",
      "                           'dmax': 100.0,\n",
      "                           'dmin': 0.0,\n",
      "                           'floor': None,\n",
      "                           'format': '{:,.1f}',\n",
      "                           'hidden': None,\n",
      "                           'max': 100,\n",
      "                           'min': 0,\n",
      "                           'minRange': None,\n",
      "                           'modify': None,\n",
      "                           'namespace': 'Samtools',\n",
      "                           'placement': 1000.0,\n",
      "                           'rid': 'mqc-generalstats-samtools-reads_mapped_percent',\n",
      "                           'scale': 'RdYlGn',\n",
      "                           'shared_key': None,\n",
      "                           'suffix': '%',\n",
      "                           'title': '% Mapped'},\n",
      "  'reads_properly_paired_percent': {'ceiling': None,\n",
      "                                    'colour': '55,126,184',\n",
      "                                    'description': '% Properly Paired Reads',\n",
      "                                    'dmax': 100.0,\n",
      "                                    'dmin': 0.0,\n",
      "                                    'floor': None,\n",
      "                                    'format': '{:,.1f}',\n",
      "                                    'hidden': False,\n",
      "                                    'max': 100,\n",
      "                                    'min': 0,\n",
      "                                    'minRange': None,\n",
      "                                    'modify': None,\n",
      "                                    'namespace': 'Samtools',\n",
      "                                    'placement': 1000.0,\n",
      "                                    'rid': 'mqc-generalstats-samtools-reads_properly_paired_percent',\n",
      "                                    'scale': 'RdYlGn',\n",
      "                                    'shared_key': None,\n",
      "                                    'suffix': '%',\n",
      "                                    'title': '% Proper Pairs'}},\n",
      " {'flagstat_total': {'ceiling': None,\n",
      "                     'colour': '77,175,74',\n",
      "                     'description': 'Total reads in the bam file (millions)',\n",
      "                     'dmax': 2.277565,\n",
      "                     'dmin': 0.0,\n",
      "                     'floor': None,\n",
      "                     'format': '{:,.1f}',\n",
      "                     'hidden': True,\n",
      "                     'max': None,\n",
      "                     'min': 0,\n",
      "                     'minRange': None,\n",
      "                     'modify': 1e-06,\n",
      "                     'namespace': 'Samtools',\n",
      "                     'placement': 1000.0,\n",
      "                     'rid': 'mqc-generalstats-samtools-flagstat_total',\n",
      "                     'scale': 'GnBu',\n",
      "                     'shared_key': 'read_count',\n",
      "                     'title': 'M Reads'},\n",
      "  'mapped_passed': {'ceiling': None,\n",
      "                    'colour': '77,175,74',\n",
      "                    'description': 'Reads Mapped in the bam file (millions)',\n",
      "                    'dmax': 2.277565,\n",
      "                    'dmin': 0.0,\n",
      "                    'floor': None,\n",
      "                    'format': '{:,.1f}',\n",
      "                    'hidden': None,\n",
      "                    'max': None,\n",
      "                    'min': 0,\n",
      "                    'minRange': None,\n",
      "                    'modify': 1e-06,\n",
      "                    'namespace': 'Samtools',\n",
      "                    'placement': 1000.0,\n",
      "                    'rid': 'mqc-generalstats-samtools-mapped_passed',\n",
      "                    'scale': 'GnBu',\n",
      "                    'shared_key': 'read_count',\n",
      "                    'title': 'M Reads Mapped'},\n",
      "  'mapped_passed_pct': {'ceiling': None,\n",
      "                        'colour': '77,175,74',\n",
      "                        'description': '% Reads Mapped in the bam file',\n",
      "                        'dmax': 100.0,\n",
      "                        'dmin': 0.0,\n",
      "                        'floor': None,\n",
      "                        'format': '{:,.1f}',\n",
      "                        'hidden': True,\n",
      "                        'max': 100,\n",
      "                        'min': 0,\n",
      "                        'minRange': None,\n",
      "                        'modify': None,\n",
      "                        'namespace': 'Samtools',\n",
      "                        'placement': 1000.0,\n",
      "                        'rid': 'mqc-generalstats-samtools-mapped_passed_pct',\n",
      "                        'scale': 'RdYlGn',\n",
      "                        'shared_key': None,\n",
      "                        'suffix': '%',\n",
      "                        'title': '% Reads Mapped'}},\n",
      " {'flagstat_total': {'ceiling': None,\n",
      "                     'colour': '77,175,74',\n",
      "                     'description': 'Total reads in the bam file (millions)',\n",
      "                     'dmax': 2.558834,\n",
      "                     'dmin': 0.0,\n",
      "                     'floor': None,\n",
      "                     'format': '{:,.1f}',\n",
      "                     'hidden': True,\n",
      "                     'max': None,\n",
      "                     'min': 0,\n",
      "                     'minRange': None,\n",
      "                     'modify': 1e-06,\n",
      "                     'namespace': 'Samtools',\n",
      "                     'placement': 1000.0,\n",
      "                     'rid': 'mqc-generalstats-samtools-flagstat_total',\n",
      "                     'scale': 'GnBu',\n",
      "                     'shared_key': 'read_count',\n",
      "                     'title': 'M Reads'},\n",
      "  'mapped_passed': {'ceiling': None,\n",
      "                    'colour': '77,175,74',\n",
      "                    'description': 'Reads Mapped in the bam file (millions)',\n",
      "                    'dmax': 2.558834,\n",
      "                    'dmin': 0.0,\n",
      "                    'floor': None,\n",
      "                    'format': '{:,.1f}',\n",
      "                    'hidden': None,\n",
      "                    'max': None,\n",
      "                    'min': 0,\n",
      "                    'minRange': None,\n",
      "                    'modify': 1e-06,\n",
      "                    'namespace': 'Samtools',\n",
      "                    'placement': 1000.0,\n",
      "                    'rid': 'mqc-generalstats-samtools-mapped_passed',\n",
      "                    'scale': 'GnBu',\n",
      "                    'shared_key': 'read_count',\n",
      "                    'title': 'M Reads Mapped'},\n",
      "  'mapped_passed_pct': {'ceiling': None,\n",
      "                        'colour': '77,175,74',\n",
      "                        'description': '% Reads Mapped in the bam file',\n",
      "                        'dmax': 100.0,\n",
      "                        'dmin': 0.0,\n",
      "                        'floor': None,\n",
      "                        'format': '{:,.1f}',\n",
      "                        'hidden': True,\n",
      "                        'max': 100,\n",
      "                        'min': 0,\n",
      "                        'minRange': None,\n",
      "                        'modify': None,\n",
      "                        'namespace': 'Samtools',\n",
      "                        'placement': 1000.0,\n",
      "                        'rid': 'mqc-generalstats-samtools-mapped_passed_pct',\n",
      "                        'scale': 'RdYlGn',\n",
      "                        'shared_key': None,\n",
      "                        'suffix': '%',\n",
      "                        'title': '% Reads Mapped'}},\n",
      " {'flagstat_total': {'ceiling': None,\n",
      "                     'colour': '77,175,74',\n",
      "                     'description': 'Total reads in the bam file (millions)',\n",
      "                     'dmax': 3.032828,\n",
      "                     'dmin': 0.0,\n",
      "                     'floor': None,\n",
      "                     'format': '{:,.1f}',\n",
      "                     'hidden': True,\n",
      "                     'max': None,\n",
      "                     'min': 0,\n",
      "                     'minRange': None,\n",
      "                     'modify': 1e-06,\n",
      "                     'namespace': 'Samtools',\n",
      "                     'placement': 1000.0,\n",
      "                     'rid': 'mqc-generalstats-samtools-flagstat_total',\n",
      "                     'scale': 'GnBu',\n",
      "                     'shared_key': 'read_count',\n",
      "                     'title': 'M Reads'},\n",
      "  'mapped_passed': {'ceiling': None,\n",
      "                    'colour': '77,175,74',\n",
      "                    'description': 'Reads Mapped in the bam file (millions)',\n",
      "                    'dmax': 3.032828,\n",
      "                    'dmin': 0.0,\n",
      "                    'floor': None,\n",
      "                    'format': '{:,.1f}',\n",
      "                    'hidden': None,\n",
      "                    'max': None,\n",
      "                    'min': 0,\n",
      "                    'minRange': None,\n",
      "                    'modify': 1e-06,\n",
      "                    'namespace': 'Samtools',\n",
      "                    'placement': 1000.0,\n",
      "                    'rid': 'mqc-generalstats-samtools-mapped_passed',\n",
      "                    'scale': 'GnBu',\n",
      "                    'shared_key': 'read_count',\n",
      "                    'title': 'M Reads Mapped'},\n",
      "  'mapped_passed_pct': {'ceiling': None,\n",
      "                        'colour': '77,175,74',\n",
      "                        'description': '% Reads Mapped in the bam file',\n",
      "                        'dmax': 100.0,\n",
      "                        'dmin': 0.0,\n",
      "                        'floor': None,\n",
      "                        'format': '{:,.1f}',\n",
      "                        'hidden': True,\n",
      "                        'max': 100,\n",
      "                        'min': 0,\n",
      "                        'minRange': None,\n",
      "                        'modify': None,\n",
      "                        'namespace': 'Samtools',\n",
      "                        'placement': 1000.0,\n",
      "                        'rid': 'mqc-generalstats-samtools-mapped_passed_pct',\n",
      "                        'scale': 'RdYlGn',\n",
      "                        'shared_key': None,\n",
      "                        'suffix': '%',\n",
      "                        'title': '% Reads Mapped'}},\n",
      " {'avg_sequence_length': {'ceiling': None,\n",
      "                          'colour': '152,78,163',\n",
      "                          'description': 'Average Read Length (bp)',\n",
      "                          'dmax': 76.0,\n",
      "                          'dmin': 0.0,\n",
      "                          'floor': None,\n",
      "                          'format': '{:,.0f}',\n",
      "                          'hidden': True,\n",
      "                          'max': None,\n",
      "                          'min': 0,\n",
      "                          'minRange': None,\n",
      "                          'modify': None,\n",
      "                          'namespace': 'FastQC',\n",
      "                          'placement': 1000.0,\n",
      "                          'rid': 'mqc-generalstats-fastqc-avg_sequence_length',\n",
      "                          'scale': 'RdYlGn',\n",
      "                          'shared_key': None,\n",
      "                          'suffix': ' bp',\n",
      "                          'title': 'Average Read Length'},\n",
      "  'median_sequence_length': {'ceiling': None,\n",
      "                             'colour': '152,78,163',\n",
      "                             'description': 'Median Read Length (bp)',\n",
      "                             'dmax': 76.0,\n",
      "                             'dmin': 0.0,\n",
      "                             'floor': None,\n",
      "                             'format': '{:,.0f}',\n",
      "                             'hidden': False,\n",
      "                             'max': None,\n",
      "                             'min': 0,\n",
      "                             'minRange': None,\n",
      "                             'modify': None,\n",
      "                             'namespace': 'FastQC',\n",
      "                             'placement': 1000.0,\n",
      "                             'rid': 'mqc-generalstats-fastqc-median_sequence_length',\n",
      "                             'scale': 'RdYlGn',\n",
      "                             'shared_key': None,\n",
      "                             'suffix': ' bp',\n",
      "                             'title': 'Median Read Length'},\n",
      "  'percent_duplicates': {'ceiling': None,\n",
      "                         'colour': '152,78,163',\n",
      "                         'description': '% Duplicate Reads',\n",
      "                         'dmax': 100.0,\n",
      "                         'dmin': 0.0,\n",
      "                         'floor': None,\n",
      "                         'format': '{:,.1f}',\n",
      "                         'hidden': None,\n",
      "                         'max': 100,\n",
      "                         'min': 0,\n",
      "                         'minRange': None,\n",
      "                         'modify': None,\n",
      "                         'namespace': 'FastQC',\n",
      "                         'placement': 1000.0,\n",
      "                         'rid': 'mqc-generalstats-fastqc-percent_duplicates',\n",
      "                         'scale': 'RdYlGn-rev',\n",
      "                         'shared_key': None,\n",
      "                         'suffix': '%',\n",
      "                         'title': '% Dups'},\n",
      "  'percent_fails': {'ceiling': None,\n",
      "                    'colour': '152,78,163',\n",
      "                    'description': 'Percentage of modules failed in FastQC '\n",
      "                                   'report (includes those not plotted here)',\n",
      "                    'dmax': 100.0,\n",
      "                    'dmin': 0.0,\n",
      "                    'floor': None,\n",
      "                    'format': '{:,.0f}',\n",
      "                    'hidden': True,\n",
      "                    'max': 100,\n",
      "                    'min': 0,\n",
      "                    'minRange': None,\n",
      "                    'modify': None,\n",
      "                    'namespace': 'FastQC',\n",
      "                    'placement': 1000.0,\n",
      "                    'rid': 'mqc-generalstats-fastqc-percent_fails',\n",
      "                    'scale': 'Reds',\n",
      "                    'shared_key': None,\n",
      "                    'suffix': '%',\n",
      "                    'title': '% Failed'},\n",
      "  'percent_gc': {'ceiling': None,\n",
      "                 'colour': '152,78,163',\n",
      "                 'description': 'Average % GC Content',\n",
      "                 'dmax': 100.0,\n",
      "                 'dmin': 0.0,\n",
      "                 'floor': None,\n",
      "                 'format': '{:,.0f}',\n",
      "                 'hidden': None,\n",
      "                 'max': 100,\n",
      "                 'min': 0,\n",
      "                 'minRange': None,\n",
      "                 'modify': None,\n",
      "                 'namespace': 'FastQC',\n",
      "                 'placement': 1000.0,\n",
      "                 'rid': 'mqc-generalstats-fastqc-percent_gc',\n",
      "                 'scale': 'Set1',\n",
      "                 'shared_key': None,\n",
      "                 'suffix': '%',\n",
      "                 'title': '% GC'},\n",
      "  'total_sequences': {'ceiling': None,\n",
      "                      'colour': '152,78,163',\n",
      "                      'description': 'Total Sequences (millions)',\n",
      "                      'dmax': 2.558834,\n",
      "                      'dmin': 0.0,\n",
      "                      'floor': None,\n",
      "                      'format': '{:,.1f}',\n",
      "                      'hidden': None,\n",
      "                      'max': None,\n",
      "                      'min': 0,\n",
      "                      'minRange': None,\n",
      "                      'modify': 1e-06,\n",
      "                      'namespace': 'FastQC',\n",
      "                      'placement': 1000.0,\n",
      "                      'rid': 'mqc-generalstats-fastqc-total_sequences',\n",
      "                      'scale': 'Blues',\n",
      "                      'shared_key': 'read_count',\n",
      "                      'title': 'M Seqs'}},\n",
      " {'avg_sequence_length': {'ceiling': None,\n",
      "                          'colour': '152,78,163',\n",
      "                          'description': 'Average Read Length (bp)',\n",
      "                          'dmax': 76.0,\n",
      "                          'dmin': 0.0,\n",
      "                          'floor': None,\n",
      "                          'format': '{:,.0f}',\n",
      "                          'hidden': True,\n",
      "                          'max': None,\n",
      "                          'min': 0,\n",
      "                          'minRange': None,\n",
      "                          'modify': None,\n",
      "                          'namespace': 'FastQC',\n",
      "                          'placement': 1000.0,\n",
      "                          'rid': 'mqc-generalstats-fastqc-avg_sequence_length',\n",
      "                          'scale': 'RdYlGn',\n",
      "                          'shared_key': None,\n",
      "                          'suffix': ' bp',\n",
      "                          'title': 'Average Read Length'},\n",
      "  'median_sequence_length': {'ceiling': None,\n",
      "                             'colour': '152,78,163',\n",
      "                             'description': 'Median Read Length (bp)',\n",
      "                             'dmax': 76.0,\n",
      "                             'dmin': 0.0,\n",
      "                             'floor': None,\n",
      "                             'format': '{:,.0f}',\n",
      "                             'hidden': True,\n",
      "                             'max': None,\n",
      "                             'min': 0,\n",
      "                             'minRange': None,\n",
      "                             'modify': None,\n",
      "                             'namespace': 'FastQC',\n",
      "                             'placement': 1000.0,\n",
      "                             'rid': 'mqc-generalstats-fastqc-median_sequence_length',\n",
      "                             'scale': 'RdYlGn',\n",
      "                             'shared_key': None,\n",
      "                             'suffix': ' bp',\n",
      "                             'title': 'Median Read Length'},\n",
      "  'percent_duplicates': {'ceiling': None,\n",
      "                         'colour': '152,78,163',\n",
      "                         'description': '% Duplicate Reads',\n",
      "                         'dmax': 100.0,\n",
      "                         'dmin': 0.0,\n",
      "                         'floor': None,\n",
      "                         'format': '{:,.1f}',\n",
      "                         'hidden': None,\n",
      "                         'max': 100,\n",
      "                         'min': 0,\n",
      "                         'minRange': None,\n",
      "                         'modify': None,\n",
      "                         'namespace': 'FastQC',\n",
      "                         'placement': 1000.0,\n",
      "                         'rid': 'mqc-generalstats-fastqc-percent_duplicates',\n",
      "                         'scale': 'RdYlGn-rev',\n",
      "                         'shared_key': None,\n",
      "                         'suffix': '%',\n",
      "                         'title': '% Dups'},\n",
      "  'percent_fails': {'ceiling': None,\n",
      "                    'colour': '152,78,163',\n",
      "                    'description': 'Percentage of modules failed in FastQC '\n",
      "                                   'report (includes those not plotted here)',\n",
      "                    'dmax': 100.0,\n",
      "                    'dmin': 0.0,\n",
      "                    'floor': None,\n",
      "                    'format': '{:,.0f}',\n",
      "                    'hidden': True,\n",
      "                    'max': 100,\n",
      "                    'min': 0,\n",
      "                    'minRange': None,\n",
      "                    'modify': None,\n",
      "                    'namespace': 'FastQC',\n",
      "                    'placement': 1000.0,\n",
      "                    'rid': 'mqc-generalstats-fastqc-percent_fails',\n",
      "                    'scale': 'Reds',\n",
      "                    'shared_key': None,\n",
      "                    'suffix': '%',\n",
      "                    'title': '% Failed'},\n",
      "  'percent_gc': {'ceiling': None,\n",
      "                 'colour': '152,78,163',\n",
      "                 'description': 'Average % GC Content',\n",
      "                 'dmax': 100.0,\n",
      "                 'dmin': 0.0,\n",
      "                 'floor': None,\n",
      "                 'format': '{:,.0f}',\n",
      "                 'hidden': None,\n",
      "                 'max': 100,\n",
      "                 'min': 0,\n",
      "                 'minRange': None,\n",
      "                 'modify': None,\n",
      "                 'namespace': 'FastQC',\n",
      "                 'placement': 1000.0,\n",
      "                 'rid': 'mqc-generalstats-fastqc-percent_gc',\n",
      "                 'scale': 'Set1',\n",
      "                 'shared_key': None,\n",
      "                 'suffix': '%',\n",
      "                 'title': '% GC'},\n",
      "  'total_sequences': {'ceiling': None,\n",
      "                      'colour': '152,78,163',\n",
      "                      'description': 'Total Sequences (millions)',\n",
      "                      'dmax': 2.277565,\n",
      "                      'dmin': 0.0,\n",
      "                      'floor': None,\n",
      "                      'format': '{:,.1f}',\n",
      "                      'hidden': None,\n",
      "                      'max': None,\n",
      "                      'min': 0,\n",
      "                      'minRange': None,\n",
      "                      'modify': 1e-06,\n",
      "                      'namespace': 'FastQC',\n",
      "                      'placement': 1000.0,\n",
      "                      'rid': 'mqc-generalstats-fastqc-total_sequences',\n",
      "                      'scale': 'Blues',\n",
      "                      'shared_key': 'read_count',\n",
      "                      'title': 'M Seqs'}},\n",
      " {'avg_sequence_length': {'ceiling': None,\n",
      "                          'colour': '152,78,163',\n",
      "                          'description': 'Average Read Length (bp)',\n",
      "                          'dmax': 76.0,\n",
      "                          'dmin': 0.0,\n",
      "                          'floor': None,\n",
      "                          'format': '{:,.0f}',\n",
      "                          'hidden': True,\n",
      "                          'max': None,\n",
      "                          'min': 0,\n",
      "                          'minRange': None,\n",
      "                          'modify': None,\n",
      "                          'namespace': 'FastQC',\n",
      "                          'placement': 1000.0,\n",
      "                          'rid': 'mqc-generalstats-fastqc-avg_sequence_length',\n",
      "                          'scale': 'RdYlGn',\n",
      "                          'shared_key': None,\n",
      "                          'suffix': ' bp',\n",
      "                          'title': 'Average Read Length'},\n",
      "  'median_sequence_length': {'ceiling': None,\n",
      "                             'colour': '152,78,163',\n",
      "                             'description': 'Median Read Length (bp)',\n",
      "                             'dmax': 76.0,\n",
      "                             'dmin': 0.0,\n",
      "                             'floor': None,\n",
      "                             'format': '{:,.0f}',\n",
      "                             'hidden': True,\n",
      "                             'max': None,\n",
      "                             'min': 0,\n",
      "                             'minRange': None,\n",
      "                             'modify': None,\n",
      "                             'namespace': 'FastQC',\n",
      "                             'placement': 1000.0,\n",
      "                             'rid': 'mqc-generalstats-fastqc-median_sequence_length',\n",
      "                             'scale': 'RdYlGn',\n",
      "                             'shared_key': None,\n",
      "                             'suffix': ' bp',\n",
      "                             'title': 'Median Read Length'},\n",
      "  'percent_duplicates': {'ceiling': None,\n",
      "                         'colour': '152,78,163',\n",
      "                         'description': '% Duplicate Reads',\n",
      "                         'dmax': 100.0,\n",
      "                         'dmin': 0.0,\n",
      "                         'floor': None,\n",
      "                         'format': '{:,.1f}',\n",
      "                         'hidden': None,\n",
      "                         'max': 100,\n",
      "                         'min': 0,\n",
      "                         'minRange': None,\n",
      "                         'modify': None,\n",
      "                         'namespace': 'FastQC',\n",
      "                         'placement': 1000.0,\n",
      "                         'rid': 'mqc-generalstats-fastqc-percent_duplicates',\n",
      "                         'scale': 'RdYlGn-rev',\n",
      "                         'shared_key': None,\n",
      "                         'suffix': '%',\n",
      "                         'title': '% Dups'},\n",
      "  'percent_fails': {'ceiling': None,\n",
      "                    'colour': '152,78,163',\n",
      "                    'description': 'Percentage of modules failed in FastQC '\n",
      "                                   'report (includes those not plotted here)',\n",
      "                    'dmax': 100.0,\n",
      "                    'dmin': 0.0,\n",
      "                    'floor': None,\n",
      "                    'format': '{:,.0f}',\n",
      "                    'hidden': True,\n",
      "                    'max': 100,\n",
      "                    'min': 0,\n",
      "                    'minRange': None,\n",
      "                    'modify': None,\n",
      "                    'namespace': 'FastQC',\n",
      "                    'placement': 1000.0,\n",
      "                    'rid': 'mqc-generalstats-fastqc-percent_fails',\n",
      "                    'scale': 'Reds',\n",
      "                    'shared_key': None,\n",
      "                    'suffix': '%',\n",
      "                    'title': '% Failed'},\n",
      "  'percent_gc': {'ceiling': None,\n",
      "                 'colour': '152,78,163',\n",
      "                 'description': 'Average % GC Content',\n",
      "                 'dmax': 100.0,\n",
      "                 'dmin': 0.0,\n",
      "                 'floor': None,\n",
      "                 'format': '{:,.0f}',\n",
      "                 'hidden': None,\n",
      "                 'max': 100,\n",
      "                 'min': 0,\n",
      "                 'minRange': None,\n",
      "                 'modify': None,\n",
      "                 'namespace': 'FastQC',\n",
      "                 'placement': 1000.0,\n",
      "                 'rid': 'mqc-generalstats-fastqc-percent_gc',\n",
      "                 'scale': 'Set1',\n",
      "                 'shared_key': None,\n",
      "                 'suffix': '%',\n",
      "                 'title': '% GC'},\n",
      "  'total_sequences': {'ceiling': None,\n",
      "                      'colour': '152,78,163',\n",
      "                      'description': 'Total Sequences (millions)',\n",
      "                      'dmax': 3.032828,\n",
      "                      'dmin': 0.0,\n",
      "                      'floor': None,\n",
      "                      'format': '{:,.1f}',\n",
      "                      'hidden': None,\n",
      "                      'max': None,\n",
      "                      'min': 0,\n",
      "                      'minRange': None,\n",
      "                      'modify': 1e-06,\n",
      "                      'namespace': 'FastQC',\n",
      "                      'placement': 1000.0,\n",
      "                      'rid': 'mqc-generalstats-fastqc-total_sequences',\n",
      "                      'scale': 'Blues',\n",
      "                      'shared_key': 'read_count',\n",
      "                      'title': 'M Seqs'}}]\n"
     ]
    }
   ],
   "source": [
    "import pymongo\n",
    "from pprint import pprint\n",
    "\n",
    "mongo_client = pymongo.MongoClient(\"mongodb://localhost:27018/\")\n",
    "mongo_db = mongo_client[\"depictioDB\"]\n",
    "\n",
    "workflow_name = \"ashleys-qc-pipeline\"\n",
    "collection = mongo_db[workflow_name]\n",
    "data_sources = collection.distinct(\"metadata.report_general_stats_headers\", {\"wf_name\": workflow_name})\n",
    "pprint(data_sources)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>description</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>suffix</th>\n",
       "      <th>scale</th>\n",
       "      <th>format</th>\n",
       "      <th>modify</th>\n",
       "      <th>namespace</th>\n",
       "      <th>rid</th>\n",
       "      <th>colour</th>\n",
       "      <th>hidden</th>\n",
       "      <th>ceiling</th>\n",
       "      <th>floor</th>\n",
       "      <th>minRange</th>\n",
       "      <th>shared_key</th>\n",
       "      <th>placement</th>\n",
       "      <th>dmax</th>\n",
       "      <th>dmin</th>\n",
       "      <th>key</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Error rate</td>\n",
       "      <td>Error rate: mismatches (NM) / bases mapped (CI...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.2f}</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-error_rate</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M Non-Primary</td>\n",
       "      <td>Non-primary alignments (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PuBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-non_primary_alignments</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>non-primary_alignments</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_mapped</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>% Mapped</td>\n",
       "      <td>% Mapped Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_mapped_percent</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>% Proper Pairs</td>\n",
       "      <td>% Properly Paired Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_properly_paire...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_properly_paired_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>% MapQ 0 Reads</td>\n",
       "      <td>% of Reads that are Ambiguously Placed (MapQ=0)</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_MQ0_percent</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_MQ0_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>M Total seqs</td>\n",
       "      <td>Total sequences in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-raw_total_sequences</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>raw_total_sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Error rate</td>\n",
       "      <td>Error rate: mismatches (NM) / bases mapped (CI...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.2f}</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-error_rate</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>M Non-Primary</td>\n",
       "      <td>Non-primary alignments (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PuBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-non_primary_alignments</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>non-primary_alignments</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_mapped</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>% Mapped</td>\n",
       "      <td>% Mapped Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_mapped_percent</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>% Proper Pairs</td>\n",
       "      <td>% Properly Paired Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_properly_paire...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_properly_paired_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>% MapQ 0 Reads</td>\n",
       "      <td>% of Reads that are Ambiguously Placed (MapQ=0)</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_MQ0_percent</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_MQ0_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>M Total seqs</td>\n",
       "      <td>Total sequences in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-raw_total_sequences</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>raw_total_sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Error rate</td>\n",
       "      <td>Error rate: mismatches (NM) / bases mapped (CI...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.2f}</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-error_rate</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>M Non-Primary</td>\n",
       "      <td>Non-primary alignments (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PuBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-non_primary_alignments</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>non-primary_alignments</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_mapped</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>% Mapped</td>\n",
       "      <td>% Mapped Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_mapped_percent</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>% Proper Pairs</td>\n",
       "      <td>% Properly Paired Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_properly_paire...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_properly_paired_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>% MapQ 0 Reads</td>\n",
       "      <td>% of Reads that are Ambiguously Placed (MapQ=0)</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-reads_MQ0_percent</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_MQ0_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>M Total seqs</td>\n",
       "      <td>Total sequences in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-raw_total_sequences</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>raw_total_sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>M Reads</td>\n",
       "      <td>Total reads in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-flagstat_total</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>flagstat_total</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-mapped_passed</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>% Reads Mapped</td>\n",
       "      <td>% Reads Mapped in the bam file</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-mapped_passed_pct</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>M Reads</td>\n",
       "      <td>Total reads in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-flagstat_total</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>flagstat_total</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-mapped_passed</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>% Reads Mapped</td>\n",
       "      <td>% Reads Mapped in the bam file</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-mapped_passed_pct</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>M Reads</td>\n",
       "      <td>Total reads in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-flagstat_total</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>flagstat_total</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-mapped_passed</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>% Reads Mapped</td>\n",
       "      <td>% Reads Mapped in the bam file</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>mqc-generalstats-samtools-mapped_passed_pct</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>% Dups</td>\n",
       "      <td>% Duplicate Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn-rev</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-percent_duplicates</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>% GC</td>\n",
       "      <td>Average % GC Content</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Set1</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-percent_gc</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_gc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Average Read Length</td>\n",
       "      <td>Average Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-avg_sequence_length</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>avg_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Median Read Length</td>\n",
       "      <td>Median Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-median_sequence_length</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>median_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>% Failed</td>\n",
       "      <td>Percentage of modules failed in FastQC report ...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Reds</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-percent_fails</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_fails</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>M Seqs</td>\n",
       "      <td>Total Sequences (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blues</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-total_sequences</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>total_sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>% Dups</td>\n",
       "      <td>% Duplicate Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn-rev</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-percent_duplicates</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>% GC</td>\n",
       "      <td>Average % GC Content</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Set1</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-percent_gc</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_gc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Average Read Length</td>\n",
       "      <td>Average Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-avg_sequence_length</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>avg_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Median Read Length</td>\n",
       "      <td>Median Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-median_sequence_length</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>median_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>% Failed</td>\n",
       "      <td>Percentage of modules failed in FastQC report ...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Reds</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-percent_fails</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_fails</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>M Seqs</td>\n",
       "      <td>Total Sequences (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blues</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-total_sequences</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>total_sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>% Dups</td>\n",
       "      <td>% Duplicate Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn-rev</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-percent_duplicates</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>% GC</td>\n",
       "      <td>Average % GC Content</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Set1</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-percent_gc</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_gc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Average Read Length</td>\n",
       "      <td>Average Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-avg_sequence_length</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>avg_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Median Read Length</td>\n",
       "      <td>Median Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-median_sequence_length</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>median_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>% Failed</td>\n",
       "      <td>Percentage of modules failed in FastQC report ...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Reds</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-percent_fails</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_fails</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>M Seqs</td>\n",
       "      <td>Total Sequences (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blues</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>mqc-generalstats-fastqc-total_sequences</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>total_sequences</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  title                                        description  \\\n",
       "0            Error rate  Error rate: mismatches (NM) / bases mapped (CI...   \n",
       "1         M Non-Primary                  Non-primary alignments (millions)   \n",
       "2        M Reads Mapped            Reads Mapped in the bam file (millions)   \n",
       "3              % Mapped                                     % Mapped Reads   \n",
       "4        % Proper Pairs                            % Properly Paired Reads   \n",
       "5        % MapQ 0 Reads    % of Reads that are Ambiguously Placed (MapQ=0)   \n",
       "6          M Total seqs         Total sequences in the bam file (millions)   \n",
       "7            Error rate  Error rate: mismatches (NM) / bases mapped (CI...   \n",
       "8         M Non-Primary                  Non-primary alignments (millions)   \n",
       "9        M Reads Mapped            Reads Mapped in the bam file (millions)   \n",
       "10             % Mapped                                     % Mapped Reads   \n",
       "11       % Proper Pairs                            % Properly Paired Reads   \n",
       "12       % MapQ 0 Reads    % of Reads that are Ambiguously Placed (MapQ=0)   \n",
       "13         M Total seqs         Total sequences in the bam file (millions)   \n",
       "14           Error rate  Error rate: mismatches (NM) / bases mapped (CI...   \n",
       "15        M Non-Primary                  Non-primary alignments (millions)   \n",
       "16       M Reads Mapped            Reads Mapped in the bam file (millions)   \n",
       "17             % Mapped                                     % Mapped Reads   \n",
       "18       % Proper Pairs                            % Properly Paired Reads   \n",
       "19       % MapQ 0 Reads    % of Reads that are Ambiguously Placed (MapQ=0)   \n",
       "20         M Total seqs         Total sequences in the bam file (millions)   \n",
       "21              M Reads             Total reads in the bam file (millions)   \n",
       "22       M Reads Mapped            Reads Mapped in the bam file (millions)   \n",
       "23       % Reads Mapped                     % Reads Mapped in the bam file   \n",
       "24              M Reads             Total reads in the bam file (millions)   \n",
       "25       M Reads Mapped            Reads Mapped in the bam file (millions)   \n",
       "26       % Reads Mapped                     % Reads Mapped in the bam file   \n",
       "27              M Reads             Total reads in the bam file (millions)   \n",
       "28       M Reads Mapped            Reads Mapped in the bam file (millions)   \n",
       "29       % Reads Mapped                     % Reads Mapped in the bam file   \n",
       "30               % Dups                                  % Duplicate Reads   \n",
       "31                 % GC                               Average % GC Content   \n",
       "32  Average Read Length                           Average Read Length (bp)   \n",
       "33   Median Read Length                            Median Read Length (bp)   \n",
       "34             % Failed  Percentage of modules failed in FastQC report ...   \n",
       "35               M Seqs                         Total Sequences (millions)   \n",
       "36               % Dups                                  % Duplicate Reads   \n",
       "37                 % GC                               Average % GC Content   \n",
       "38  Average Read Length                           Average Read Length (bp)   \n",
       "39   Median Read Length                            Median Read Length (bp)   \n",
       "40             % Failed  Percentage of modules failed in FastQC report ...   \n",
       "41               M Seqs                         Total Sequences (millions)   \n",
       "42               % Dups                                  % Duplicate Reads   \n",
       "43                 % GC                               Average % GC Content   \n",
       "44  Average Read Length                           Average Read Length (bp)   \n",
       "45   Median Read Length                            Median Read Length (bp)   \n",
       "46             % Failed  Percentage of modules failed in FastQC report ...   \n",
       "47               M Seqs                         Total Sequences (millions)   \n",
       "\n",
       "    min    max suffix       scale   format      modify namespace  \\\n",
       "0     0  100.0      %        OrRd  {:,.2f}  100.000000  Samtools   \n",
       "1     0    NaN    NaN        PuBu  {:,.1f}    0.000001  Samtools   \n",
       "2     0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "3     0  100.0      %      RdYlGn  {:,.1f}         NaN  Samtools   \n",
       "4     0  100.0      %      RdYlGn  {:,.1f}         NaN  Samtools   \n",
       "5     0  100.0      %        OrRd  {:,.1f}         NaN  Samtools   \n",
       "6     0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "7     0  100.0      %        OrRd  {:,.2f}  100.000000  Samtools   \n",
       "8     0    NaN    NaN        PuBu  {:,.1f}    0.000001  Samtools   \n",
       "9     0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "10    0  100.0      %      RdYlGn  {:,.1f}         NaN  Samtools   \n",
       "11    0  100.0      %      RdYlGn  {:,.1f}         NaN  Samtools   \n",
       "12    0  100.0      %        OrRd  {:,.1f}         NaN  Samtools   \n",
       "13    0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "14    0  100.0      %        OrRd  {:,.2f}  100.000000  Samtools   \n",
       "15    0    NaN    NaN        PuBu  {:,.1f}    0.000001  Samtools   \n",
       "16    0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "17    0  100.0      %      RdYlGn  {:,.1f}         NaN  Samtools   \n",
       "18    0  100.0      %      RdYlGn  {:,.1f}         NaN  Samtools   \n",
       "19    0  100.0      %        OrRd  {:,.1f}         NaN  Samtools   \n",
       "20    0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "21    0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "22    0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "23    0  100.0      %      RdYlGn  {:,.1f}         NaN  Samtools   \n",
       "24    0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "25    0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "26    0  100.0      %      RdYlGn  {:,.1f}         NaN  Samtools   \n",
       "27    0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "28    0    NaN    NaN        GnBu  {:,.1f}    0.000001  Samtools   \n",
       "29    0  100.0      %      RdYlGn  {:,.1f}         NaN  Samtools   \n",
       "30    0  100.0      %  RdYlGn-rev  {:,.1f}         NaN    FastQC   \n",
       "31    0  100.0      %        Set1  {:,.0f}         NaN    FastQC   \n",
       "32    0    NaN     bp      RdYlGn  {:,.0f}         NaN    FastQC   \n",
       "33    0    NaN     bp      RdYlGn  {:,.0f}         NaN    FastQC   \n",
       "34    0  100.0      %        Reds  {:,.0f}         NaN    FastQC   \n",
       "35    0    NaN    NaN       Blues  {:,.1f}    0.000001    FastQC   \n",
       "36    0  100.0      %  RdYlGn-rev  {:,.1f}         NaN    FastQC   \n",
       "37    0  100.0      %        Set1  {:,.0f}         NaN    FastQC   \n",
       "38    0    NaN     bp      RdYlGn  {:,.0f}         NaN    FastQC   \n",
       "39    0    NaN     bp      RdYlGn  {:,.0f}         NaN    FastQC   \n",
       "40    0  100.0      %        Reds  {:,.0f}         NaN    FastQC   \n",
       "41    0    NaN    NaN       Blues  {:,.1f}    0.000001    FastQC   \n",
       "42    0  100.0      %  RdYlGn-rev  {:,.1f}         NaN    FastQC   \n",
       "43    0  100.0      %        Set1  {:,.0f}         NaN    FastQC   \n",
       "44    0    NaN     bp      RdYlGn  {:,.0f}         NaN    FastQC   \n",
       "45    0    NaN     bp      RdYlGn  {:,.0f}         NaN    FastQC   \n",
       "46    0  100.0      %        Reds  {:,.0f}         NaN    FastQC   \n",
       "47    0    NaN    NaN       Blues  {:,.1f}    0.000001    FastQC   \n",
       "\n",
       "                                                  rid      colour hidden  \\\n",
       "0                mqc-generalstats-samtools-error_rate  55,126,184   None   \n",
       "1    mqc-generalstats-samtools-non_primary_alignments  55,126,184   None   \n",
       "2              mqc-generalstats-samtools-reads_mapped  55,126,184   None   \n",
       "3      mqc-generalstats-samtools-reads_mapped_percent  55,126,184   None   \n",
       "4   mqc-generalstats-samtools-reads_properly_paire...  55,126,184  False   \n",
       "5         mqc-generalstats-samtools-reads_MQ0_percent  55,126,184   True   \n",
       "6       mqc-generalstats-samtools-raw_total_sequences  55,126,184   None   \n",
       "7                mqc-generalstats-samtools-error_rate  55,126,184   None   \n",
       "8    mqc-generalstats-samtools-non_primary_alignments  55,126,184   None   \n",
       "9              mqc-generalstats-samtools-reads_mapped  55,126,184   None   \n",
       "10     mqc-generalstats-samtools-reads_mapped_percent  55,126,184   None   \n",
       "11  mqc-generalstats-samtools-reads_properly_paire...  55,126,184  False   \n",
       "12        mqc-generalstats-samtools-reads_MQ0_percent  55,126,184   True   \n",
       "13      mqc-generalstats-samtools-raw_total_sequences  55,126,184   None   \n",
       "14               mqc-generalstats-samtools-error_rate  55,126,184   None   \n",
       "15   mqc-generalstats-samtools-non_primary_alignments  55,126,184   None   \n",
       "16             mqc-generalstats-samtools-reads_mapped  55,126,184   None   \n",
       "17     mqc-generalstats-samtools-reads_mapped_percent  55,126,184   None   \n",
       "18  mqc-generalstats-samtools-reads_properly_paire...  55,126,184  False   \n",
       "19        mqc-generalstats-samtools-reads_MQ0_percent  55,126,184   True   \n",
       "20      mqc-generalstats-samtools-raw_total_sequences  55,126,184   None   \n",
       "21           mqc-generalstats-samtools-flagstat_total   77,175,74   True   \n",
       "22            mqc-generalstats-samtools-mapped_passed   77,175,74   None   \n",
       "23        mqc-generalstats-samtools-mapped_passed_pct   77,175,74   True   \n",
       "24           mqc-generalstats-samtools-flagstat_total   77,175,74   True   \n",
       "25            mqc-generalstats-samtools-mapped_passed   77,175,74   None   \n",
       "26        mqc-generalstats-samtools-mapped_passed_pct   77,175,74   True   \n",
       "27           mqc-generalstats-samtools-flagstat_total   77,175,74   True   \n",
       "28            mqc-generalstats-samtools-mapped_passed   77,175,74   None   \n",
       "29        mqc-generalstats-samtools-mapped_passed_pct   77,175,74   True   \n",
       "30         mqc-generalstats-fastqc-percent_duplicates  152,78,163   None   \n",
       "31                 mqc-generalstats-fastqc-percent_gc  152,78,163   None   \n",
       "32        mqc-generalstats-fastqc-avg_sequence_length  152,78,163   True   \n",
       "33     mqc-generalstats-fastqc-median_sequence_length  152,78,163  False   \n",
       "34              mqc-generalstats-fastqc-percent_fails  152,78,163   True   \n",
       "35            mqc-generalstats-fastqc-total_sequences  152,78,163   None   \n",
       "36         mqc-generalstats-fastqc-percent_duplicates  152,78,163   None   \n",
       "37                 mqc-generalstats-fastqc-percent_gc  152,78,163   None   \n",
       "38        mqc-generalstats-fastqc-avg_sequence_length  152,78,163   True   \n",
       "39     mqc-generalstats-fastqc-median_sequence_length  152,78,163   True   \n",
       "40              mqc-generalstats-fastqc-percent_fails  152,78,163   True   \n",
       "41            mqc-generalstats-fastqc-total_sequences  152,78,163   None   \n",
       "42         mqc-generalstats-fastqc-percent_duplicates  152,78,163   None   \n",
       "43                 mqc-generalstats-fastqc-percent_gc  152,78,163   None   \n",
       "44        mqc-generalstats-fastqc-avg_sequence_length  152,78,163   True   \n",
       "45     mqc-generalstats-fastqc-median_sequence_length  152,78,163   True   \n",
       "46              mqc-generalstats-fastqc-percent_fails  152,78,163   True   \n",
       "47            mqc-generalstats-fastqc-total_sequences  152,78,163   None   \n",
       "\n",
       "   ceiling floor minRange  shared_key  placement        dmax  dmin  \\\n",
       "0     None  None     None        None     1000.0  100.000000   0.0   \n",
       "1     None  None     None  read_count     1000.0    2.277565   0.0   \n",
       "2     None  None     None  read_count     1000.0    2.277565   0.0   \n",
       "3     None  None     None        None     1000.0  100.000000   0.0   \n",
       "4     None  None     None        None     1000.0  100.000000   0.0   \n",
       "5     None  None     None        None     1000.0  100.000000   0.0   \n",
       "6     None  None     None  read_count     1000.0    2.277565   0.0   \n",
       "7     None  None     None        None     1000.0  100.000000   0.0   \n",
       "8     None  None     None  read_count     1000.0    2.558834   0.0   \n",
       "9     None  None     None  read_count     1000.0    2.558834   0.0   \n",
       "10    None  None     None        None     1000.0  100.000000   0.0   \n",
       "11    None  None     None        None     1000.0  100.000000   0.0   \n",
       "12    None  None     None        None     1000.0  100.000000   0.0   \n",
       "13    None  None     None  read_count     1000.0    2.558834   0.0   \n",
       "14    None  None     None        None     1000.0  100.000000   0.0   \n",
       "15    None  None     None  read_count     1000.0    3.032828   0.0   \n",
       "16    None  None     None  read_count     1000.0    3.032828   0.0   \n",
       "17    None  None     None        None     1000.0  100.000000   0.0   \n",
       "18    None  None     None        None     1000.0  100.000000   0.0   \n",
       "19    None  None     None        None     1000.0  100.000000   0.0   \n",
       "20    None  None     None  read_count     1000.0    3.032828   0.0   \n",
       "21    None  None     None  read_count     1000.0    2.277565   0.0   \n",
       "22    None  None     None  read_count     1000.0    2.277565   0.0   \n",
       "23    None  None     None        None     1000.0  100.000000   0.0   \n",
       "24    None  None     None  read_count     1000.0    2.558834   0.0   \n",
       "25    None  None     None  read_count     1000.0    2.558834   0.0   \n",
       "26    None  None     None        None     1000.0  100.000000   0.0   \n",
       "27    None  None     None  read_count     1000.0    3.032828   0.0   \n",
       "28    None  None     None  read_count     1000.0    3.032828   0.0   \n",
       "29    None  None     None        None     1000.0  100.000000   0.0   \n",
       "30    None  None     None        None     1000.0  100.000000   0.0   \n",
       "31    None  None     None        None     1000.0  100.000000   0.0   \n",
       "32    None  None     None        None     1000.0   76.000000   0.0   \n",
       "33    None  None     None        None     1000.0   76.000000   0.0   \n",
       "34    None  None     None        None     1000.0  100.000000   0.0   \n",
       "35    None  None     None  read_count     1000.0    2.558834   0.0   \n",
       "36    None  None     None        None     1000.0  100.000000   0.0   \n",
       "37    None  None     None        None     1000.0  100.000000   0.0   \n",
       "38    None  None     None        None     1000.0   76.000000   0.0   \n",
       "39    None  None     None        None     1000.0   76.000000   0.0   \n",
       "40    None  None     None        None     1000.0  100.000000   0.0   \n",
       "41    None  None     None  read_count     1000.0    2.277565   0.0   \n",
       "42    None  None     None        None     1000.0  100.000000   0.0   \n",
       "43    None  None     None        None     1000.0  100.000000   0.0   \n",
       "44    None  None     None        None     1000.0   76.000000   0.0   \n",
       "45    None  None     None        None     1000.0   76.000000   0.0   \n",
       "46    None  None     None        None     1000.0  100.000000   0.0   \n",
       "47    None  None     None  read_count     1000.0    3.032828   0.0   \n",
       "\n",
       "                              key  \n",
       "0                      error_rate  \n",
       "1          non-primary_alignments  \n",
       "2                    reads_mapped  \n",
       "3            reads_mapped_percent  \n",
       "4   reads_properly_paired_percent  \n",
       "5               reads_MQ0_percent  \n",
       "6             raw_total_sequences  \n",
       "7                      error_rate  \n",
       "8          non-primary_alignments  \n",
       "9                    reads_mapped  \n",
       "10           reads_mapped_percent  \n",
       "11  reads_properly_paired_percent  \n",
       "12              reads_MQ0_percent  \n",
       "13            raw_total_sequences  \n",
       "14                     error_rate  \n",
       "15         non-primary_alignments  \n",
       "16                   reads_mapped  \n",
       "17           reads_mapped_percent  \n",
       "18  reads_properly_paired_percent  \n",
       "19              reads_MQ0_percent  \n",
       "20            raw_total_sequences  \n",
       "21                 flagstat_total  \n",
       "22                  mapped_passed  \n",
       "23              mapped_passed_pct  \n",
       "24                 flagstat_total  \n",
       "25                  mapped_passed  \n",
       "26              mapped_passed_pct  \n",
       "27                 flagstat_total  \n",
       "28                  mapped_passed  \n",
       "29              mapped_passed_pct  \n",
       "30             percent_duplicates  \n",
       "31                     percent_gc  \n",
       "32            avg_sequence_length  \n",
       "33         median_sequence_length  \n",
       "34                  percent_fails  \n",
       "35                total_sequences  \n",
       "36             percent_duplicates  \n",
       "37                     percent_gc  \n",
       "38            avg_sequence_length  \n",
       "39         median_sequence_length  \n",
       "40                  percent_fails  \n",
       "41                total_sequences  \n",
       "42             percent_duplicates  \n",
       "43                     percent_gc  \n",
       "44            avg_sequence_length  \n",
       "45         median_sequence_length  \n",
       "46                  percent_fails  \n",
       "47                total_sequences  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.DataFrame.from_records([d[key] | {\"key\": key} for d in data_sources for key in d])\n",
    "df.groupby(\"namespace\")[\"description\"].unique().to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'FastQC': array(['% Duplicate Reads', 'Average % GC Content',\n",
       "        'Average Read Length (bp)', 'Median Read Length (bp)',\n",
       "        'Percentage of modules failed in FastQC report (includes those not plotted here)',\n",
       "        'Total Sequences (millions)'], dtype=object),\n",
       " 'Samtools': array(['Error rate: mismatches (NM) / bases mapped (CIGAR)',\n",
       "        'Non-primary alignments (millions)',\n",
       "        'Reads Mapped in the bam file (millions)', '% Mapped Reads',\n",
       "        '% Properly Paired Reads',\n",
       "        '% of Reads that are Ambiguously Placed (MapQ=0)',\n",
       "        'Total sequences in the bam file (millions)',\n",
       "        'Total reads in the bam file (millions)',\n",
       "        '% Reads Mapped in the bam file'], dtype=object)}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.groupby(\"namespace\")[\"description\"].unique().to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pymongo.results.InsertManyResult at 0x7fa8b5c57dc0>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dev_col = mongo_db[\"dev\"]\n",
    "dev_col.insert_many(df.to_dict(\"records\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_id</th>\n",
       "      <th>title</th>\n",
       "      <th>description</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>suffix</th>\n",
       "      <th>scale</th>\n",
       "      <th>format</th>\n",
       "      <th>modify</th>\n",
       "      <th>namespace</th>\n",
       "      <th>...</th>\n",
       "      <th>colour</th>\n",
       "      <th>hidden</th>\n",
       "      <th>ceiling</th>\n",
       "      <th>floor</th>\n",
       "      <th>minRange</th>\n",
       "      <th>shared_key</th>\n",
       "      <th>placement</th>\n",
       "      <th>dmax</th>\n",
       "      <th>dmin</th>\n",
       "      <th>key</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6438737175e1b1fe946f8a75</td>\n",
       "      <td>Error rate</td>\n",
       "      <td>Error rate: mismatches (NM) / bases mapped (CI...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.2f}</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6438737175e1b1fe946f8a76</td>\n",
       "      <td>M Non-Primary</td>\n",
       "      <td>Non-primary alignments (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PuBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>non-primary_alignments</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6438737175e1b1fe946f8a77</td>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6438737175e1b1fe946f8a78</td>\n",
       "      <td>% Mapped</td>\n",
       "      <td>% Mapped Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6438737175e1b1fe946f8a79</td>\n",
       "      <td>% Proper Pairs</td>\n",
       "      <td>% Properly Paired Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_properly_paired_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6438737175e1b1fe946f8a7a</td>\n",
       "      <td>% MapQ 0 Reads</td>\n",
       "      <td>% of Reads that are Ambiguously Placed (MapQ=0)</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_MQ0_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6438737175e1b1fe946f8a7b</td>\n",
       "      <td>M Total seqs</td>\n",
       "      <td>Total sequences in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>raw_total_sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6438737175e1b1fe946f8a7c</td>\n",
       "      <td>Error rate</td>\n",
       "      <td>Error rate: mismatches (NM) / bases mapped (CI...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.2f}</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>6438737175e1b1fe946f8a7d</td>\n",
       "      <td>M Non-Primary</td>\n",
       "      <td>Non-primary alignments (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PuBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>non-primary_alignments</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>6438737175e1b1fe946f8a7e</td>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>6438737175e1b1fe946f8a7f</td>\n",
       "      <td>% Mapped</td>\n",
       "      <td>% Mapped Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>6438737175e1b1fe946f8a80</td>\n",
       "      <td>% Proper Pairs</td>\n",
       "      <td>% Properly Paired Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_properly_paired_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>6438737175e1b1fe946f8a81</td>\n",
       "      <td>% MapQ 0 Reads</td>\n",
       "      <td>% of Reads that are Ambiguously Placed (MapQ=0)</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_MQ0_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>6438737175e1b1fe946f8a82</td>\n",
       "      <td>M Total seqs</td>\n",
       "      <td>Total sequences in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>raw_total_sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>6438737175e1b1fe946f8a83</td>\n",
       "      <td>Error rate</td>\n",
       "      <td>Error rate: mismatches (NM) / bases mapped (CI...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.2f}</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>6438737175e1b1fe946f8a84</td>\n",
       "      <td>M Non-Primary</td>\n",
       "      <td>Non-primary alignments (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PuBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>non-primary_alignments</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>6438737175e1b1fe946f8a85</td>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6438737175e1b1fe946f8a86</td>\n",
       "      <td>% Mapped</td>\n",
       "      <td>% Mapped Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_mapped_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>6438737175e1b1fe946f8a87</td>\n",
       "      <td>% Proper Pairs</td>\n",
       "      <td>% Properly Paired Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_properly_paired_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>6438737175e1b1fe946f8a88</td>\n",
       "      <td>% MapQ 0 Reads</td>\n",
       "      <td>% of Reads that are Ambiguously Placed (MapQ=0)</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>OrRd</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>reads_MQ0_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>6438737175e1b1fe946f8a89</td>\n",
       "      <td>M Total seqs</td>\n",
       "      <td>Total sequences in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>55,126,184</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>raw_total_sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>6438737175e1b1fe946f8a8a</td>\n",
       "      <td>M Reads</td>\n",
       "      <td>Total reads in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>flagstat_total</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>6438737175e1b1fe946f8a8b</td>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>6438737175e1b1fe946f8a8c</td>\n",
       "      <td>% Reads Mapped</td>\n",
       "      <td>% Reads Mapped in the bam file</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>6438737175e1b1fe946f8a8d</td>\n",
       "      <td>M Reads</td>\n",
       "      <td>Total reads in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>flagstat_total</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>6438737175e1b1fe946f8a8e</td>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>6438737175e1b1fe946f8a8f</td>\n",
       "      <td>% Reads Mapped</td>\n",
       "      <td>% Reads Mapped in the bam file</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>6438737175e1b1fe946f8a90</td>\n",
       "      <td>M Reads</td>\n",
       "      <td>Total reads in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>flagstat_total</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>6438737175e1b1fe946f8a91</td>\n",
       "      <td>M Reads Mapped</td>\n",
       "      <td>Reads Mapped in the bam file (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GnBu</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>6438737175e1b1fe946f8a92</td>\n",
       "      <td>% Reads Mapped</td>\n",
       "      <td>% Reads Mapped in the bam file</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Samtools</td>\n",
       "      <td>...</td>\n",
       "      <td>77,175,74</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mapped_passed_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>6438737175e1b1fe946f8a93</td>\n",
       "      <td>% Dups</td>\n",
       "      <td>% Duplicate Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn-rev</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>6438737175e1b1fe946f8a94</td>\n",
       "      <td>% GC</td>\n",
       "      <td>Average % GC Content</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Set1</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_gc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>6438737175e1b1fe946f8a95</td>\n",
       "      <td>Average Read Length</td>\n",
       "      <td>Average Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>avg_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>6438737175e1b1fe946f8a96</td>\n",
       "      <td>Median Read Length</td>\n",
       "      <td>Median Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>median_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>6438737175e1b1fe946f8a97</td>\n",
       "      <td>% Failed</td>\n",
       "      <td>Percentage of modules failed in FastQC report ...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Reds</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_fails</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>6438737175e1b1fe946f8a98</td>\n",
       "      <td>M Seqs</td>\n",
       "      <td>Total Sequences (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blues</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.558834</td>\n",
       "      <td>0.0</td>\n",
       "      <td>total_sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>6438737175e1b1fe946f8a99</td>\n",
       "      <td>% Dups</td>\n",
       "      <td>% Duplicate Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn-rev</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>6438737175e1b1fe946f8a9a</td>\n",
       "      <td>% GC</td>\n",
       "      <td>Average % GC Content</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Set1</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_gc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>6438737175e1b1fe946f8a9b</td>\n",
       "      <td>Average Read Length</td>\n",
       "      <td>Average Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>avg_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>6438737175e1b1fe946f8a9c</td>\n",
       "      <td>Median Read Length</td>\n",
       "      <td>Median Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>median_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>6438737175e1b1fe946f8a9d</td>\n",
       "      <td>% Failed</td>\n",
       "      <td>Percentage of modules failed in FastQC report ...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Reds</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_fails</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>6438737175e1b1fe946f8a9e</td>\n",
       "      <td>M Seqs</td>\n",
       "      <td>Total Sequences (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blues</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>2.277565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>total_sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>6438737175e1b1fe946f8a9f</td>\n",
       "      <td>% Dups</td>\n",
       "      <td>% Duplicate Reads</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>RdYlGn-rev</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>6438737175e1b1fe946f8aa0</td>\n",
       "      <td>% GC</td>\n",
       "      <td>Average % GC Content</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Set1</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_gc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>6438737175e1b1fe946f8aa1</td>\n",
       "      <td>Average Read Length</td>\n",
       "      <td>Average Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>avg_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>6438737175e1b1fe946f8aa2</td>\n",
       "      <td>Median Read Length</td>\n",
       "      <td>Median Read Length (bp)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bp</td>\n",
       "      <td>RdYlGn</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>76.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>median_sequence_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>6438737175e1b1fe946f8aa3</td>\n",
       "      <td>% Failed</td>\n",
       "      <td>Percentage of modules failed in FastQC report ...</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>%</td>\n",
       "      <td>Reds</td>\n",
       "      <td>{:,.0f}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>percent_fails</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>6438737175e1b1fe946f8aa4</td>\n",
       "      <td>M Seqs</td>\n",
       "      <td>Total Sequences (millions)</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blues</td>\n",
       "      <td>{:,.1f}</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>FastQC</td>\n",
       "      <td>...</td>\n",
       "      <td>152,78,163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>read_count</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>3.032828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>total_sequences</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         _id                title  \\\n",
       "0   6438737175e1b1fe946f8a75           Error rate   \n",
       "1   6438737175e1b1fe946f8a76        M Non-Primary   \n",
       "2   6438737175e1b1fe946f8a77       M Reads Mapped   \n",
       "3   6438737175e1b1fe946f8a78             % Mapped   \n",
       "4   6438737175e1b1fe946f8a79       % Proper Pairs   \n",
       "5   6438737175e1b1fe946f8a7a       % MapQ 0 Reads   \n",
       "6   6438737175e1b1fe946f8a7b         M Total seqs   \n",
       "7   6438737175e1b1fe946f8a7c           Error rate   \n",
       "8   6438737175e1b1fe946f8a7d        M Non-Primary   \n",
       "9   6438737175e1b1fe946f8a7e       M Reads Mapped   \n",
       "10  6438737175e1b1fe946f8a7f             % Mapped   \n",
       "11  6438737175e1b1fe946f8a80       % Proper Pairs   \n",
       "12  6438737175e1b1fe946f8a81       % MapQ 0 Reads   \n",
       "13  6438737175e1b1fe946f8a82         M Total seqs   \n",
       "14  6438737175e1b1fe946f8a83           Error rate   \n",
       "15  6438737175e1b1fe946f8a84        M Non-Primary   \n",
       "16  6438737175e1b1fe946f8a85       M Reads Mapped   \n",
       "17  6438737175e1b1fe946f8a86             % Mapped   \n",
       "18  6438737175e1b1fe946f8a87       % Proper Pairs   \n",
       "19  6438737175e1b1fe946f8a88       % MapQ 0 Reads   \n",
       "20  6438737175e1b1fe946f8a89         M Total seqs   \n",
       "21  6438737175e1b1fe946f8a8a              M Reads   \n",
       "22  6438737175e1b1fe946f8a8b       M Reads Mapped   \n",
       "23  6438737175e1b1fe946f8a8c       % Reads Mapped   \n",
       "24  6438737175e1b1fe946f8a8d              M Reads   \n",
       "25  6438737175e1b1fe946f8a8e       M Reads Mapped   \n",
       "26  6438737175e1b1fe946f8a8f       % Reads Mapped   \n",
       "27  6438737175e1b1fe946f8a90              M Reads   \n",
       "28  6438737175e1b1fe946f8a91       M Reads Mapped   \n",
       "29  6438737175e1b1fe946f8a92       % Reads Mapped   \n",
       "30  6438737175e1b1fe946f8a93               % Dups   \n",
       "31  6438737175e1b1fe946f8a94                 % GC   \n",
       "32  6438737175e1b1fe946f8a95  Average Read Length   \n",
       "33  6438737175e1b1fe946f8a96   Median Read Length   \n",
       "34  6438737175e1b1fe946f8a97             % Failed   \n",
       "35  6438737175e1b1fe946f8a98               M Seqs   \n",
       "36  6438737175e1b1fe946f8a99               % Dups   \n",
       "37  6438737175e1b1fe946f8a9a                 % GC   \n",
       "38  6438737175e1b1fe946f8a9b  Average Read Length   \n",
       "39  6438737175e1b1fe946f8a9c   Median Read Length   \n",
       "40  6438737175e1b1fe946f8a9d             % Failed   \n",
       "41  6438737175e1b1fe946f8a9e               M Seqs   \n",
       "42  6438737175e1b1fe946f8a9f               % Dups   \n",
       "43  6438737175e1b1fe946f8aa0                 % GC   \n",
       "44  6438737175e1b1fe946f8aa1  Average Read Length   \n",
       "45  6438737175e1b1fe946f8aa2   Median Read Length   \n",
       "46  6438737175e1b1fe946f8aa3             % Failed   \n",
       "47  6438737175e1b1fe946f8aa4               M Seqs   \n",
       "\n",
       "                                          description  min    max suffix  \\\n",
       "0   Error rate: mismatches (NM) / bases mapped (CI...    0  100.0      %   \n",
       "1                   Non-primary alignments (millions)    0    NaN    NaN   \n",
       "2             Reads Mapped in the bam file (millions)    0    NaN    NaN   \n",
       "3                                      % Mapped Reads    0  100.0      %   \n",
       "4                             % Properly Paired Reads    0  100.0      %   \n",
       "5     % of Reads that are Ambiguously Placed (MapQ=0)    0  100.0      %   \n",
       "6          Total sequences in the bam file (millions)    0    NaN    NaN   \n",
       "7   Error rate: mismatches (NM) / bases mapped (CI...    0  100.0      %   \n",
       "8                   Non-primary alignments (millions)    0    NaN    NaN   \n",
       "9             Reads Mapped in the bam file (millions)    0    NaN    NaN   \n",
       "10                                     % Mapped Reads    0  100.0      %   \n",
       "11                            % Properly Paired Reads    0  100.0      %   \n",
       "12    % of Reads that are Ambiguously Placed (MapQ=0)    0  100.0      %   \n",
       "13         Total sequences in the bam file (millions)    0    NaN    NaN   \n",
       "14  Error rate: mismatches (NM) / bases mapped (CI...    0  100.0      %   \n",
       "15                  Non-primary alignments (millions)    0    NaN    NaN   \n",
       "16            Reads Mapped in the bam file (millions)    0    NaN    NaN   \n",
       "17                                     % Mapped Reads    0  100.0      %   \n",
       "18                            % Properly Paired Reads    0  100.0      %   \n",
       "19    % of Reads that are Ambiguously Placed (MapQ=0)    0  100.0      %   \n",
       "20         Total sequences in the bam file (millions)    0    NaN    NaN   \n",
       "21             Total reads in the bam file (millions)    0    NaN    NaN   \n",
       "22            Reads Mapped in the bam file (millions)    0    NaN    NaN   \n",
       "23                     % Reads Mapped in the bam file    0  100.0      %   \n",
       "24             Total reads in the bam file (millions)    0    NaN    NaN   \n",
       "25            Reads Mapped in the bam file (millions)    0    NaN    NaN   \n",
       "26                     % Reads Mapped in the bam file    0  100.0      %   \n",
       "27             Total reads in the bam file (millions)    0    NaN    NaN   \n",
       "28            Reads Mapped in the bam file (millions)    0    NaN    NaN   \n",
       "29                     % Reads Mapped in the bam file    0  100.0      %   \n",
       "30                                  % Duplicate Reads    0  100.0      %   \n",
       "31                               Average % GC Content    0  100.0      %   \n",
       "32                           Average Read Length (bp)    0    NaN     bp   \n",
       "33                            Median Read Length (bp)    0    NaN     bp   \n",
       "34  Percentage of modules failed in FastQC report ...    0  100.0      %   \n",
       "35                         Total Sequences (millions)    0    NaN    NaN   \n",
       "36                                  % Duplicate Reads    0  100.0      %   \n",
       "37                               Average % GC Content    0  100.0      %   \n",
       "38                           Average Read Length (bp)    0    NaN     bp   \n",
       "39                            Median Read Length (bp)    0    NaN     bp   \n",
       "40  Percentage of modules failed in FastQC report ...    0  100.0      %   \n",
       "41                         Total Sequences (millions)    0    NaN    NaN   \n",
       "42                                  % Duplicate Reads    0  100.0      %   \n",
       "43                               Average % GC Content    0  100.0      %   \n",
       "44                           Average Read Length (bp)    0    NaN     bp   \n",
       "45                            Median Read Length (bp)    0    NaN     bp   \n",
       "46  Percentage of modules failed in FastQC report ...    0  100.0      %   \n",
       "47                         Total Sequences (millions)    0    NaN    NaN   \n",
       "\n",
       "         scale   format      modify namespace  ...      colour hidden ceiling  \\\n",
       "0         OrRd  {:,.2f}  100.000000  Samtools  ...  55,126,184   None    None   \n",
       "1         PuBu  {:,.1f}    0.000001  Samtools  ...  55,126,184   None    None   \n",
       "2         GnBu  {:,.1f}    0.000001  Samtools  ...  55,126,184   None    None   \n",
       "3       RdYlGn  {:,.1f}         NaN  Samtools  ...  55,126,184   None    None   \n",
       "4       RdYlGn  {:,.1f}         NaN  Samtools  ...  55,126,184  False    None   \n",
       "5         OrRd  {:,.1f}         NaN  Samtools  ...  55,126,184   True    None   \n",
       "6         GnBu  {:,.1f}    0.000001  Samtools  ...  55,126,184   None    None   \n",
       "7         OrRd  {:,.2f}  100.000000  Samtools  ...  55,126,184   None    None   \n",
       "8         PuBu  {:,.1f}    0.000001  Samtools  ...  55,126,184   None    None   \n",
       "9         GnBu  {:,.1f}    0.000001  Samtools  ...  55,126,184   None    None   \n",
       "10      RdYlGn  {:,.1f}         NaN  Samtools  ...  55,126,184   None    None   \n",
       "11      RdYlGn  {:,.1f}         NaN  Samtools  ...  55,126,184  False    None   \n",
       "12        OrRd  {:,.1f}         NaN  Samtools  ...  55,126,184   True    None   \n",
       "13        GnBu  {:,.1f}    0.000001  Samtools  ...  55,126,184   None    None   \n",
       "14        OrRd  {:,.2f}  100.000000  Samtools  ...  55,126,184   None    None   \n",
       "15        PuBu  {:,.1f}    0.000001  Samtools  ...  55,126,184   None    None   \n",
       "16        GnBu  {:,.1f}    0.000001  Samtools  ...  55,126,184   None    None   \n",
       "17      RdYlGn  {:,.1f}         NaN  Samtools  ...  55,126,184   None    None   \n",
       "18      RdYlGn  {:,.1f}         NaN  Samtools  ...  55,126,184  False    None   \n",
       "19        OrRd  {:,.1f}         NaN  Samtools  ...  55,126,184   True    None   \n",
       "20        GnBu  {:,.1f}    0.000001  Samtools  ...  55,126,184   None    None   \n",
       "21        GnBu  {:,.1f}    0.000001  Samtools  ...   77,175,74   True    None   \n",
       "22        GnBu  {:,.1f}    0.000001  Samtools  ...   77,175,74   None    None   \n",
       "23      RdYlGn  {:,.1f}         NaN  Samtools  ...   77,175,74   True    None   \n",
       "24        GnBu  {:,.1f}    0.000001  Samtools  ...   77,175,74   True    None   \n",
       "25        GnBu  {:,.1f}    0.000001  Samtools  ...   77,175,74   None    None   \n",
       "26      RdYlGn  {:,.1f}         NaN  Samtools  ...   77,175,74   True    None   \n",
       "27        GnBu  {:,.1f}    0.000001  Samtools  ...   77,175,74   True    None   \n",
       "28        GnBu  {:,.1f}    0.000001  Samtools  ...   77,175,74   None    None   \n",
       "29      RdYlGn  {:,.1f}         NaN  Samtools  ...   77,175,74   True    None   \n",
       "30  RdYlGn-rev  {:,.1f}         NaN    FastQC  ...  152,78,163   None    None   \n",
       "31        Set1  {:,.0f}         NaN    FastQC  ...  152,78,163   None    None   \n",
       "32      RdYlGn  {:,.0f}         NaN    FastQC  ...  152,78,163   True    None   \n",
       "33      RdYlGn  {:,.0f}         NaN    FastQC  ...  152,78,163  False    None   \n",
       "34        Reds  {:,.0f}         NaN    FastQC  ...  152,78,163   True    None   \n",
       "35       Blues  {:,.1f}    0.000001    FastQC  ...  152,78,163   None    None   \n",
       "36  RdYlGn-rev  {:,.1f}         NaN    FastQC  ...  152,78,163   None    None   \n",
       "37        Set1  {:,.0f}         NaN    FastQC  ...  152,78,163   None    None   \n",
       "38      RdYlGn  {:,.0f}         NaN    FastQC  ...  152,78,163   True    None   \n",
       "39      RdYlGn  {:,.0f}         NaN    FastQC  ...  152,78,163   True    None   \n",
       "40        Reds  {:,.0f}         NaN    FastQC  ...  152,78,163   True    None   \n",
       "41       Blues  {:,.1f}    0.000001    FastQC  ...  152,78,163   None    None   \n",
       "42  RdYlGn-rev  {:,.1f}         NaN    FastQC  ...  152,78,163   None    None   \n",
       "43        Set1  {:,.0f}         NaN    FastQC  ...  152,78,163   None    None   \n",
       "44      RdYlGn  {:,.0f}         NaN    FastQC  ...  152,78,163   True    None   \n",
       "45      RdYlGn  {:,.0f}         NaN    FastQC  ...  152,78,163   True    None   \n",
       "46        Reds  {:,.0f}         NaN    FastQC  ...  152,78,163   True    None   \n",
       "47       Blues  {:,.1f}    0.000001    FastQC  ...  152,78,163   None    None   \n",
       "\n",
       "   floor minRange  shared_key placement        dmax  dmin  \\\n",
       "0   None     None        None    1000.0  100.000000   0.0   \n",
       "1   None     None  read_count    1000.0    2.277565   0.0   \n",
       "2   None     None  read_count    1000.0    2.277565   0.0   \n",
       "3   None     None        None    1000.0  100.000000   0.0   \n",
       "4   None     None        None    1000.0  100.000000   0.0   \n",
       "5   None     None        None    1000.0  100.000000   0.0   \n",
       "6   None     None  read_count    1000.0    2.277565   0.0   \n",
       "7   None     None        None    1000.0  100.000000   0.0   \n",
       "8   None     None  read_count    1000.0    2.558834   0.0   \n",
       "9   None     None  read_count    1000.0    2.558834   0.0   \n",
       "10  None     None        None    1000.0  100.000000   0.0   \n",
       "11  None     None        None    1000.0  100.000000   0.0   \n",
       "12  None     None        None    1000.0  100.000000   0.0   \n",
       "13  None     None  read_count    1000.0    2.558834   0.0   \n",
       "14  None     None        None    1000.0  100.000000   0.0   \n",
       "15  None     None  read_count    1000.0    3.032828   0.0   \n",
       "16  None     None  read_count    1000.0    3.032828   0.0   \n",
       "17  None     None        None    1000.0  100.000000   0.0   \n",
       "18  None     None        None    1000.0  100.000000   0.0   \n",
       "19  None     None        None    1000.0  100.000000   0.0   \n",
       "20  None     None  read_count    1000.0    3.032828   0.0   \n",
       "21  None     None  read_count    1000.0    2.277565   0.0   \n",
       "22  None     None  read_count    1000.0    2.277565   0.0   \n",
       "23  None     None        None    1000.0  100.000000   0.0   \n",
       "24  None     None  read_count    1000.0    2.558834   0.0   \n",
       "25  None     None  read_count    1000.0    2.558834   0.0   \n",
       "26  None     None        None    1000.0  100.000000   0.0   \n",
       "27  None     None  read_count    1000.0    3.032828   0.0   \n",
       "28  None     None  read_count    1000.0    3.032828   0.0   \n",
       "29  None     None        None    1000.0  100.000000   0.0   \n",
       "30  None     None        None    1000.0  100.000000   0.0   \n",
       "31  None     None        None    1000.0  100.000000   0.0   \n",
       "32  None     None        None    1000.0   76.000000   0.0   \n",
       "33  None     None        None    1000.0   76.000000   0.0   \n",
       "34  None     None        None    1000.0  100.000000   0.0   \n",
       "35  None     None  read_count    1000.0    2.558834   0.0   \n",
       "36  None     None        None    1000.0  100.000000   0.0   \n",
       "37  None     None        None    1000.0  100.000000   0.0   \n",
       "38  None     None        None    1000.0   76.000000   0.0   \n",
       "39  None     None        None    1000.0   76.000000   0.0   \n",
       "40  None     None        None    1000.0  100.000000   0.0   \n",
       "41  None     None  read_count    1000.0    2.277565   0.0   \n",
       "42  None     None        None    1000.0  100.000000   0.0   \n",
       "43  None     None        None    1000.0  100.000000   0.0   \n",
       "44  None     None        None    1000.0   76.000000   0.0   \n",
       "45  None     None        None    1000.0   76.000000   0.0   \n",
       "46  None     None        None    1000.0  100.000000   0.0   \n",
       "47  None     None  read_count    1000.0    3.032828   0.0   \n",
       "\n",
       "                              key  \n",
       "0                      error_rate  \n",
       "1          non-primary_alignments  \n",
       "2                    reads_mapped  \n",
       "3            reads_mapped_percent  \n",
       "4   reads_properly_paired_percent  \n",
       "5               reads_MQ0_percent  \n",
       "6             raw_total_sequences  \n",
       "7                      error_rate  \n",
       "8          non-primary_alignments  \n",
       "9                    reads_mapped  \n",
       "10           reads_mapped_percent  \n",
       "11  reads_properly_paired_percent  \n",
       "12              reads_MQ0_percent  \n",
       "13            raw_total_sequences  \n",
       "14                     error_rate  \n",
       "15         non-primary_alignments  \n",
       "16                   reads_mapped  \n",
       "17           reads_mapped_percent  \n",
       "18  reads_properly_paired_percent  \n",
       "19              reads_MQ0_percent  \n",
       "20            raw_total_sequences  \n",
       "21                 flagstat_total  \n",
       "22                  mapped_passed  \n",
       "23              mapped_passed_pct  \n",
       "24                 flagstat_total  \n",
       "25                  mapped_passed  \n",
       "26              mapped_passed_pct  \n",
       "27                 flagstat_total  \n",
       "28                  mapped_passed  \n",
       "29              mapped_passed_pct  \n",
       "30             percent_duplicates  \n",
       "31                     percent_gc  \n",
       "32            avg_sequence_length  \n",
       "33         median_sequence_length  \n",
       "34                  percent_fails  \n",
       "35                total_sequences  \n",
       "36             percent_duplicates  \n",
       "37                     percent_gc  \n",
       "38            avg_sequence_length  \n",
       "39         median_sequence_length  \n",
       "40                  percent_fails  \n",
       "41                total_sequences  \n",
       "42             percent_duplicates  \n",
       "43                     percent_gc  \n",
       "44            avg_sequence_length  \n",
       "45         median_sequence_length  \n",
       "46                  percent_fails  \n",
       "47                total_sequences  \n",
       "\n",
       "[48 rows x 21 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Query data from MongoDB and convert it to Arrow table\n",
    "cursor = dev_col.find({})\n",
    "df = pd.DataFrame(list(cursor))\n",
    "\n",
    "# Convert Arrow table back to pandas DataFrame\n",
    "df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "depictio",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
